auth/api/iam/openid4vp.go

```diff
@@ -549,43 +548,40 @@ func (r Wrapper) handleAccessTokenRequest(ctx context.Context, verifier did.DID,
 	if request.ClientId == nil {
 		return nil, oauthError(oauth.InvalidRequest, "missing client_id parameter")
 	}
-	// get initial session using state parameter
-	var serverSession OAuthSession
-	err := r.oauthServerStateStore().Get(*request.State, &serverSession)
-	if err != nil {
-		return nil, oauthError(oauth.InvalidRequest, "invalid or expired state")
-	}
-	// check if the code_verifier is valid
-	serverSession.PKCEParams.Verifier = *request.CodeVerifier
-	if !validatePKCEParams(serverSession.PKCEParams) {
-		return nil, oauthError(oauth.InvalidRequest, "invalid code_verifier")
-	}
 	// check if the authorization code is valid
 	var oauthSession OAuthSession
-	err = r.oauthCodeStore().Get(*request.Code, &oauthSession)
+	err := r.oauthCodeStore().Get(*request.Code, &oauthSession)
```

Comment:
Encrypting the PKCE params into the authz code seems like unnecessary complexity if the `oauthSession` here (`handleAccessTokenRequest`) contains the values anyway. Is there a benefit I'm missing?

Reply:
awww crap, I thought there was a lack of server state from the authorise to the token calls, but it's passed via the nonce and code stores.....